pkg/awsutils/awssession/session.go

```diff
@@ -65,6 +66,21 @@ func New() *session.Session {
 		},
 		STSRegionalEndpoint: endpoints.RegionalSTSEndpoint,
 	}
+
+	endpoint := os.Getenv("AWS_EC2_ENDPOINT")
+	if endpoint != "" {
+		customResolver := func(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
+			if service == ec2.EndpointsID {
+				return endpoints.ResolvedEndpoint{
+					URL:           endpoint,
+					SigningRegion: region,
```

Comment:
Is there any value in the region being configurable?

Reply:
Not really. `region` will be empty string. I think it should be removed. `amazon-vpc-cni-k8s` configures the `region` variable outside of `awssession.New()`, there is no need to set the variable.

Reply:
I removed `region` variable in the latest commit.